modules/navier_stokes/include/auxkernels/v2fViscosityAux.h

```diff
@@ -0,0 +1,44 @@
+
+
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "AuxKernel.h"
+
+/**
+ * Computes the turbuent viscosity for the v2f model.
+ */
+class v2fViscosityAux : public AuxKernel
```

Comment:
My preference would be to default to creating functor materials instead of aux kernels. We lost derivative information recently with the `WCNSFV2PSlipVelocityAux` that was leading to poor convergence in tests that used a Newton solve type. Even if you never intend to use a Newton solve type with this object, I don't see a disadvantage to creating the functor material object instead. If one wants to maybe save some on-the-fly evaluations you can always use a `FunctorAux` to populate an aux variable from the functor material.

Reply:
Thanks. Added an `INSFVv2fViscosityFunctorMaterial`. I am keeping the AuxKernel since it may still be useful later for the linear solvers.

Reply:
Is that for time derivatives? That is the only use case I see for the aux kernel. One could write a lambda or helper template that is used by Real and ADReal functor material properties. The Real property would be useful for linear solvers while the ADReal property would be useful for nonlinear solvers

Reply:
Sorry that I am just seeing this. Yes. We can do that. How is the evaluation time of the AuxKernel compared to the a Real material? If it is equally fast, then I am all for that

Reply:
A single evaluation of a Real functor material property and an aux kernel should have identical speeds. If we're worried about multiple evaluations of the functor material property slowing things down and there is not a concern about dropping derivatives because you're not using Newton or, then that would a motivation for using a `FunctorAux` to "save-off" the material property evaluation. The functor material property-`FunctorAux` pairing would also satisfy the need for time derivatives as well if that's ever needed

Reply:
I have been trying this out. There are two things that I don't like much about putting everything in a material. First, I am not very happy on the workflow `FunctorMaterial` -> `FunctorAux` -> `AuxVariable`, when we could just have an `AuxKernel` populating the `AuxVariable`. Then, I expect that for more complicated modifications on top of v2f, there will be some capping and activation of the kernel that will be iteration dependent. This will be easier to do on the `AuxKernel` instead of the material. Is it alright if we keep the `FunctorMaterial` and the `AuxKernel` for now?

Reply:
> I am not very happy on the workflow FunctorMaterial -> FunctorAux -> AuxVariable

that s not the official workflow anymore.
use `output_material_property = ..` parameter instead on the Material

Reply:
> Then, I expect that for more complicated modifications on top of v2f, there will be some capping and activation of the kernel that will be iteration dependent

What would be easier in the aux kernel? Functors allow evaluation with arbitrary states now, e.g. previous time-step, previous nonlinear/linear iteration.

I just know that in your rayleigh-bernard PR tests, convergence was so much better once I converted an aux kernel to functor material property such that the Jacobian was correct for Newton. If you wanted to go with the aux kernel, then I think I'd ask for you to retrieve the solve type from the Executioner and error if the solve type is Newton.